nsxt/resource_nsxt_policy_tier0_gateway_ha_vip_config.go

```diff
@@ -0,0 +1,364 @@
+/* Copyright © 2019 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	gm_tier0s "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-gm/global_infra/tier_0s"
+	gm_model "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-gm/model"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/tier_0s"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"strconv"
+	"strings"
+)
+
+var interfacePathLen = 8
+var interfaceTier0Location = 3
+var interfaceLocaleSrvLocation = 5
+
+func resourceNsxtPolicyTier0GatewayHAVipConfig() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyTier0GatewayHAVipConfigCreate,
+		Read:   resourceNsxtPolicyTier0GatewayHAVipConfigRead,
+		Update: resourceNsxtPolicyTier0GatewayHAVipConfigUpdate,
+		Delete: resourceNsxtPolicyTier0GatewayHAVipConfigDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceNsxtPolicyTier0GatewayHAVipConfigImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"config": {
+				Type:        schema.TypeList,
+				Description: "Tier0 HA VIP Config",
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:        schema.TypeBool,
+							Description: "Flag to enable this HA VIP config",
+							Optional:    true,
+							Default:     true,
+						},
+						"external_interface_paths": {
+							Type:        schema.TypeList,
+							Description: "paths to Tier0 external interfaces which are to be paired to provide redundancy",
+							Required:    true,
+							MinItems:    2,
+							MaxItems:    2,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validatePolicyPath(),
+							},
+						},
+						"vip_subnets": {
+							Type:        schema.TypeList,
+							Description: "IP address subnets which will be used as floating IP addresses",
+							Required:    true,
+							MinItems:    1,
+							MaxItems:    2,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validateIPCidr(),
+							},
+						},
+					},
+				},
+			},
+			"locale_service_id": {
```

Comment:
we expose locale service path, so I think path should be used here? This attribute is for sake of dependencies I think (otherwise it can be deferred from interface path), so it would make sense to use path.

Reply:
Actually I don't think we need this at all, since interfaces are already dependent on GW

Reply:
Its here just because it is useful in the code to have it for update/read/delete.
We can hide it from the documentation if you think it is not needed there.

Reply:
Sorry my bad, didn't notice its computed.

Reply:
However, I think in this case locale_service_id should be inside the config clause, as it should be possible to have separate configs for separate locale services on GM.

Reply:
The way it currently work is that all the entries in the same resource config have to match the same locale-service. If the use wants another vip-config for a different locale-service, it will be in a different resource.
I think it is simpler this way. I'll update the doc to reflect it.

Reply:
But is it possible to have more than one pair of HA VIP within same locale service?

Reply:
Yep I see its possible to create 2 pairs. I'm trying to think how to name the 'config' section so its less confusong.. maybe vip_pair? Anyway we can rename it later.